configure.ac

```diff
@@ -1544,10 +1544,10 @@ case $host_os in
 esac
 
 ## Windows
-case "`uname`" in
-  MINGW*)
+case "$host_os" in
+  *mingw*)
     # The Winsock library
-    AC_CHECK_LIB([ws2_32], [GetUserName])
+    AC_CHECK_LIB([ws2_32], [htons])
```

Comment:
GetUserName() doesn't really appear in Winsock2 (it's an alias for GetUserNameA or GetUserNameEx) and isn't used in the library anyway.